.github/workflows/publish-extractor-lib.yml

```diff
@@ -19,14 +19,14 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
 
+      - name: Install Poetry
+        run: pipx install poetry
+
       - name: Install Python 3.9
-        uses: actions/setup-python@3105fb18c05ddd93efea5f9e0bef7a03a6e9e7df
+        uses: actions/setup-python@98f2ad02fd48d057ee3b4d4f66525b231c3e52b6 # v3.1.2
         with:
           python-version: '3.9.x'
```

Comment:
SHOULD: The `cache` input is missing here.

Reply:
Do you mean I should add 

`- name: Setup Dependency Caching
        uses: actions/cache@c64c572235d810460d0d6876e9c705ad5002b353
        id: cache-venv
        with:
          path: ${{ steps.venv-path.outputs.dir }}
          key: ${{ runner.os }}-poetry-${{ hashFiles(env.LOCK_FILE) }}
          restore-keys: |
            ${{ runner.os }}-poetry-`

Below this line in the code?